### ISSUE TITLE
bug(build): Python basic image is too large

### ISSUE THREAD

**Issue by gaocegege** [tensorchord/envd#1086]
## Description

The CPU image built by envd is about 1GB. I am wondering if our LLB works well. It is too large.

## Additional Info

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by aseaday** [comment]
Maybe you mean pytorch image?

**Comment by gaocegege** [comment]
https://github.com/tensorchord/envd/blob/main/examples/python-basic/build.envd This one.

**Comment by gaocegege** [comment]
Ref https://github.com/zhuwenxing/envd_demo/actions/runs/3361616233

Here is a reproducible case about the build speed of envd and docker

**Comment by popfido** [comment]
If it is a development environment image (which is what `envd`'s target) I think it is quite reasonable?

**Comment by popfido** [comment]
If it is a production environment image I may think it would be a quite large one.

**Comment by popfido** [comment]
An idea I would like to host in issues is that does `envd` try to do only as a development environment manager for general AI for science teams? Or it would also taking exporting to production environment as a further option? (I did not check the roadmap at the time I leave this message)

**Comment by kemingy** [comment]
Some related issues:

* #1176

**Comment by popfido** [comment]
> > An idea I would like to host in issues is that does `envd` try to do only as a development environment manager for general AI for science teams? Or it would also taking exporting to production environment as a further option? (I did not check the roadmap at the time I leave this message)
> 
> * It will [feat(lang): Support serving #157](https://github.com/tensorchord/envd/issues/157)
> 
> BTW, ML production images can be huge (up to 10GB).

Yeah, it's a case by case problem. But in production images we just need minimal requirements for runtime(when codes for production are stable), which means it should usually be smaller than its corresponding development images. 

But for development images, we require a high flexibility to meet debug/develop functionality. Which obviously needs more tool installations and optional configurations.
